git_greppers/processing.sh

```
@@ -0,0 +1,281 @@
+        echo "#KEY#EXCLUDES# pnpm.lock, TypeScript build files, Rust build files" >> "$current_file"
+        echo "" >> "$current_file"
+      fi
+      
+      echo "#KEY#PROCESS_COMMIT# $processed_count/$commit_count $commit_date - $commit_msg" >> "$log_file"
+      debug 1 "Appending to diff file: $current_file"
+      
+      # Print progress with unbuffered output
+      printf "${CYAN}[%s/%s] Processing:${NC} %s - %s\n" "$processed_count" "$commit_count" "$commit_date" "$commit_msg"
+      
+      # Create diff section header
+      echo "#KEY#START_COMMIT#" >> "$current_file"
```

Shellcheck (0.10.0)

[style] Consider using { cmd1; cmd2; } >> file instead of individual redirects.

(SC2129)